### DISCUSSION TITLE
the method "helper.api.stix_cyber_observable.read" in the connector.

### DISCUSSION THREAD

**Discussion by Jijach** [OpenCTI-Platform/opencti#6880] [Q&A] [answered]
Hi, I want to write my own connector, but I'm faced with the problem of getting data from the platform. I am trying to get only observable with type "ipv4-addr" as follows:

        _val = ["ipv4-addr"]
        opencti_entity = self.helper.api.stix_cyber_observable.read(
            filters={
                "mode":"and",
                "filters":[{
                    "key":"type",
                    "mode":"or",
                    "values":_val
                }],
                "filterGroups":[],
            }
        )  
        
but there is no result, only None. I tried to set other keys, but I get an error:
{'name': 'UNSUPPORTED_ERROR', 'message': 'incorrect filter keys not existing in any schema definition'}

What don't I understand? I will be grateful for any help.

**Comment by yassine-ouaamou** [comment, marked as answer]
Hi @Jijach,
To filter on the entity type, use the key "entity_type" and to get all the observables corresponding to your filter, use list():
```
observables = opencti_api_client.stix_cyber_observable.list(
    filters={
        "mode":"and",
        "filters":[{
            "key":"entity_type",
            "mode":"or",
            "operator": "eq",
            "values":["IPv4-Addr"]
        }],
        "filterGroups":[],
    }
)
```

**Reply by Jijach** [reply]
I have already found a workaround for my problem, but thank you so much for this information!
